src/main/java/bio/terra/service/job/JobService.java

```diff
@@ -245,28 +246,33 @@ public JobModel mapFlightStateToJobModel(FlightState flightState) {
       completedDate = flightState.getCompleted().get().toString();
     }
 
-    JobModel jobModel =
-        new JobModel()
-            .id(flightState.getFlightId())
-            .description(description)
-            .jobStatus(jobStatus)
-            .statusCode(statusCode.value())
-            .submitted(submittedDate)
-            .completed(completedDate);
-
-    return jobModel;
+    return new JobModel()
+        .id(flightState.getFlightId())
+        .description(description)
+        .jobStatus(jobStatus)
+        .statusCode(statusCode.value())
+        .submitted(submittedDate)
+        .completed(completedDate);
   }
 
-  private JobModel.JobStatusEnum getJobStatus(FlightStatus flightStatus) {
+  private JobModel.JobStatusEnum getJobStatus(FlightState flightState) {
+    FlightStatus flightStatus = flightState.getFlightStatus();
     switch (flightStatus) {
       case ERROR:
-        return JobModel.JobStatusEnum.FAILED;
       case FATAL:
         return JobModel.JobStatusEnum.FAILED;
       case RUNNING:
         return JobModel.JobStatusEnum.RUNNING;
       case SUCCESS:
-        return JobModel.JobStatusEnum.SUCCEEDED;
+        boolean succeededWithWarnings =
+            flightState
+                .getResultMap()
+                .filter(rm -> rm.containsKey(CommonFlightKeys.FLIGHT_HAS_WARNINGS))
+                .map(rm -> rm.get(CommonFlightKeys.FLIGHT_HAS_WARNINGS, Boolean.class))
+                .orElse(false);
+        return succeededWithWarnings
```

Comment:
I thought we were planning on having this return success and then have a field in the result include a boolean indicating that there was a warning?

Reply:
We had chatted about that, but this given a clearer indication of the result of a job even without getting the result. Users can make decisions based on the terminal job status, and the result can remain unchanged. Plus, this means not changing the different types of bulk load responses and offloading that responsibility onto the user.

Reply:
I guess I'm nervous about injecting a new kind of response code...it feels wrong.  Isn't this what the results endpoint is for?

Reply:
I think it makes sense here. The flight succeeded, but we're telling the user that there are caveats to that success. The JobResult can then be inspected to see that more than the max allowed failures occurred. Returning `succeeded_with_warnings` allows people to write code that branches off of one API call rather than two.

Reply:
I guess I maintain that this is similar behavior to db bulk loaders which return success regardless of how many rows were ignored for constrain violations.  I feel like folks should always be checking the result...even if files loaded, maybe it's not what the users expected, etc.